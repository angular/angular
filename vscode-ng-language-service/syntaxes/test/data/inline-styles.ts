/* clang-format off */

@Component({
//// Property key/value test
  styles: [ '.example { width: 100px; }' ],

//// Multiple styles test
  styles: [
    '.example { width: 100px; }',
    '.example { height: 100px; }',
  ],

//// String delimiter tests
  styles: [ `.example { width: 100px; }` ],
  styles: [ ".example { width: 100px; }" ],
  styles: [ '.example { width: 100px; }' ],

//// Parenthesization tests
  styles: ( (( [ ( '.example { width: 100px; }' ) ] )) ),
//// styles string
  styles: ( ((  ( '.example { width: 100px; }' )  )) ),
  styles:  `.example { width: 100px; }` ,
  styles:  ".example { width: 100px; }" ,
  styles:  '.example { width: 100px; }' ,
})
export class TMComponent{}
