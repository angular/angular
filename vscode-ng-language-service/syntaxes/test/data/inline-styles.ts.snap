>/* clang-format off */
#^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng
>
>@Component({
#^^^^^^^^^^^^^ inline-styles.ng
>//// Property key/value test
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng
>  styles: [ '.example { width: 100px; }' ],
#^^ inline-styles.ng
#  ^^^^^^ inline-styles.ng meta.object-literal.key.ts
#        ^ inline-styles.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#         ^ inline-styles.ng
#          ^ inline-styles.ng meta.array.literal.ts meta.brace.square.ts
#           ^ inline-styles.ng
#            ^ inline-styles.ng string
#             ^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng source.css.scss
#                                       ^ inline-styles.ng string
#                                        ^ inline-styles.ng
#                                         ^ inline-styles.ng meta.array.literal.ts meta.brace.square.ts
#                                          ^^ inline-styles.ng
>
>//// Multiple styles test
#^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng
>  styles: [
#^^ inline-styles.ng
#  ^^^^^^ inline-styles.ng meta.object-literal.key.ts
#        ^ inline-styles.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#         ^ inline-styles.ng
#          ^ inline-styles.ng meta.array.literal.ts meta.brace.square.ts
>    '.example { width: 100px; }',
#^^^^ inline-styles.ng
#    ^ inline-styles.ng string
#     ^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng source.css.scss
#                               ^ inline-styles.ng string
#                                ^^ inline-styles.ng
>    '.example { height: 100px; }',
#^^^^ inline-styles.ng
#    ^ inline-styles.ng string
#     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng source.css.scss
#                                ^ inline-styles.ng string
#                                 ^^ inline-styles.ng
>  ],
#^^ inline-styles.ng
#  ^ inline-styles.ng meta.array.literal.ts meta.brace.square.ts
#   ^^ inline-styles.ng
>
>//// String delimiter tests
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng
>  styles: [ `.example { width: 100px; }` ],
#^^ inline-styles.ng
#  ^^^^^^ inline-styles.ng meta.object-literal.key.ts
#        ^ inline-styles.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#         ^ inline-styles.ng
#          ^ inline-styles.ng meta.array.literal.ts meta.brace.square.ts
#           ^ inline-styles.ng
#            ^ inline-styles.ng string
#             ^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng source.css.scss
#                                       ^ inline-styles.ng string
#                                        ^ inline-styles.ng
#                                         ^ inline-styles.ng meta.array.literal.ts meta.brace.square.ts
#                                          ^^ inline-styles.ng
>  styles: [ ".example { width: 100px; }" ],
#^^ inline-styles.ng
#  ^^^^^^ inline-styles.ng meta.object-literal.key.ts
#        ^ inline-styles.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#         ^ inline-styles.ng
#          ^ inline-styles.ng meta.array.literal.ts meta.brace.square.ts
#           ^ inline-styles.ng
#            ^ inline-styles.ng string
#             ^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng source.css.scss
#                                       ^ inline-styles.ng string
#                                        ^ inline-styles.ng
#                                         ^ inline-styles.ng meta.array.literal.ts meta.brace.square.ts
#                                          ^^ inline-styles.ng
>  styles: [ '.example { width: 100px; }' ],
#^^ inline-styles.ng
#  ^^^^^^ inline-styles.ng meta.object-literal.key.ts
#        ^ inline-styles.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#         ^ inline-styles.ng
#          ^ inline-styles.ng meta.array.literal.ts meta.brace.square.ts
#           ^ inline-styles.ng
#            ^ inline-styles.ng string
#             ^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng source.css.scss
#                                       ^ inline-styles.ng string
#                                        ^ inline-styles.ng
#                                         ^ inline-styles.ng meta.array.literal.ts meta.brace.square.ts
#                                          ^^ inline-styles.ng
>
>//// Parenthesization tests
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng
>  styles: ( (( [ ( '.example { width: 100px; }' ) ] )) ),
#^^ inline-styles.ng
#  ^^^^^^ inline-styles.ng meta.object-literal.key.ts
#        ^ inline-styles.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#         ^ inline-styles.ng
#          ^ inline-styles.ng meta.brace.round.ts
#           ^^^^^^^ inline-styles.ng
#                  ^ inline-styles.ng
#                   ^ inline-styles.ng string
#                    ^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng source.css.scss
#                                              ^ inline-styles.ng string
#                                               ^ inline-styles.ng
#                                                ^ inline-styles.ng meta.brace.round.ts
#                                                 ^^^^^^^ inline-styles.ng
#                                                        ^^ inline-styles.ng
>//// styles string
#^^^^^^^^^^^^^^^^^^^ inline-styles.ng
>  styles: ( ((  ( '.example { width: 100px; }' )  )) ),
#^^ inline-styles.ng
#  ^^^^^^ inline-styles.ng meta.object-literal.key.ts
#        ^ inline-styles.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#         ^ inline-styles.ng
#          ^ inline-styles.ng meta.brace.round.ts
#           ^^^^^^ inline-styles.ng
#                 ^ inline-styles.ng
#                  ^ inline-styles.ng string
#                   ^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng source.css.scss
#                                             ^ inline-styles.ng string
#                                              ^ inline-styles.ng
#                                               ^ inline-styles.ng meta.brace.round.ts
#                                                ^^^^^^ inline-styles.ng
#                                                      ^^ inline-styles.ng
>  styles:  `.example { width: 100px; }` ,
#^^ inline-styles.ng
#  ^^^^^^ inline-styles.ng meta.object-literal.key.ts
#        ^ inline-styles.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#         ^^ inline-styles.ng
#           ^ inline-styles.ng string
#            ^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng source.css.scss
#                                      ^ inline-styles.ng string
#                                       ^ inline-styles.ng
#                                        ^^ inline-styles.ng
>  styles:  ".example { width: 100px; }" ,
#^^ inline-styles.ng
#  ^^^^^^ inline-styles.ng meta.object-literal.key.ts
#        ^ inline-styles.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#         ^^ inline-styles.ng
#           ^ inline-styles.ng string
#            ^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng source.css.scss
#                                      ^ inline-styles.ng string
#                                       ^ inline-styles.ng
#                                        ^^ inline-styles.ng
>  styles:  '.example { width: 100px; }' ,
#^^ inline-styles.ng
#  ^^^^^^ inline-styles.ng meta.object-literal.key.ts
#        ^ inline-styles.ng meta.object-literal.key.ts punctuation.separator.key-value.ts
#         ^^ inline-styles.ng
#           ^ inline-styles.ng string
#            ^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng source.css.scss
#                                      ^ inline-styles.ng string
#                                       ^ inline-styles.ng
#                                        ^^ inline-styles.ng
>})
#^^^ inline-styles.ng
>export class TMComponent{}
#^^^^^^^^^^^^^^^^^^^^^^^^^^^ inline-styles.ng
>